istioctl/cmd/auth.go

```diff
@@ -102,41 +103,47 @@ THIS COMMAND IS STILL UNDER ACTIVE DEVELOPMENT AND NOT READY FOR PRODUCTION USE.
 		},
 	}
 
-	upgradeCmd = &cobra.Command{
-		Hidden: true,
-		Use:    "upgrade",
-		Short:  "Upgrade Istio Authorization Policy from version v1alpha1 to v1beta1",
-		Long: `Upgrade converts Istio authorization policy from version v1alpha1 to v1beta1. It requires access to Kubernetes
-service definition in order to translate the service name specified in the ServiceRole to the corresponding
-workload labels in the AuthorizationPolicy. The service definition could be provided either from the current
-Kubernetes cluster or from a YAML file specified from command line.
+	convertCmd = &cobra.Command{
+		Use:   "convert",
+		Short: "Convert v1alpha1 RBAC policy to v1beta1 authorization policy",
+		Long: `Convert converts Istio v1alpha1 RBAC policy to v1beta1 authorization policy. The command talks to Kubernetes
+API server to get all the information needed to complete the conversion, including the currently applied v1alpha1
+RBAC policies, the Istio config-map for root namespace configuration and the k8s Service translating the
+service name to workload selector.
+
+The tool could also be used in offline mode without talking to the Kubernetes API server. In this mode,
+all needed information are provided though command line.
+
+The conversion result is printed to the stdout, you can take a final review of the converted policies
+before applying it.
 
 THIS COMMAND IS STILL UNDER ACTIVE DEVELOPMENT AND NOT READY FOR PRODUCTION USE.
 `,
-		Example: `  # Upgrade the Istio authorization policy with service definition from the current Kubernetes cluster:
-  istioctl experimental auth upgrade -f rbac-v1-policy.yaml,rbac-v1-policy-2.yaml`,
+		Example: `  # Convert the v1alpha1 RBAC policy applied in the current cluster:
+  istioctl experimental auth convert
```

Comment:
Can we change `auth` to `authz` instead? I remember from our previous conversion that this tool can be used for authn conversion, but since this tool now can check the applied policies on the cluster, I feel like users may want to have a way to convert either authN or authZ only. Only, both cases do not share a lot of code, except maybe the workload selector finding.

Reply:
Yeah I'm also thinking this, it's better to rename this to `authz` and make it to focus on the authz policies. But I prefer to do it in a separate PR for cleaning all related information in istioctl.

Reply:
sgtm